### ISSUE TITLE
specify output filename for deduplication

### ISSUE THREAD

**Issue by davidries84** [FelixKrueger/Bismark#238]
Hey,

short: 
could deduplicate_bismark get a -o option to set the name of the output file?

Long:

I'm using the deduplication step to merge bam files, that belong to one sample.
I have files from multiple sequencing runs and they are named accordingly.
The problem is, that the outputfile is named depending on the first input file, and
this is somewhat random. It would be best, if I could name the output like the input sample.

Best, 

David

**Comment by FelixKrueger** [comment]
Hi David,

I have added a new option `-o/--outfile` to deduplicate_bismark.

I implemented it so that you specify a basename for the desired output file. This basename is then modified to remove file endings such as `.bam`, `.sam`, `.txt` or `.gz` , and `.deduplicated.bam`, or `.multiple.deduplicated.bam` in `--multiple` mode, is then appended for consistency reasons. I hope this is workable for you.

While I was at it, I also made `--bam` the default mode. Uncompressed SAM output may be obtained using the new option `--sam`. Let me know if this solves your problems.
